File: 2015/day01/java/Day01P1.java
//2025 Levi D. Smith
import java.io.*;

public class Day01P1 {
  public static void main(String[] args) {

    try {
      BufferedReader br = new BufferedReader(new FileReader("../input01.txt"));
      String strLine;

      while ((strLine = br.readLine()) != null) {

        int iValue = 0;
//        System.out.println("line: " + strLine);
	int i;
	char[] chars = strLine.toCharArray();
	for (i = 0; i < chars.length; i++) {
          switch(chars[i]) {
            case '(':
              iValue++;
	      break;
	    case ')':
	      iValue--;
	      break;
          }
	}
        System.out.println(String.valueOf(iValue));
      }
      br.close();
    } catch (IOException e) {
      System.out.println("File Not Found");
    }

  }
}

File: 2015/day01/ruby/day01p1.rb
#2025 Levi D. Smith
f = open("../input01.txt")
while( strLine = f.gets )
  iValue = 0
  strLine.chars.each { 
    | c |
    case c[0]
    when '('
      iValue += 1 
    when ')'
      iValue -= 1
    end
  }
  puts iValue
end
f.close
